lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py

```diff
@@ -1014,6 +1040,9 @@ def exec_module(self, **kwargs):
                                                                      parsed_availability_set.get('name'))
                         availability_set_resource = self.compute_models.SubResource(id=availability_set.id)
 
+                        if self.zones:
+                            self.fail("Parameter error: you can't use Availability Set and Availability Zones at the same time")
```

Comment:
I think we should have warning instead of fail here, but i think it can be done later. no need to hold this pr.